standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -7223,13 +7214,23 @@ private List<Partition> get_partitions_by_filter_internal(final String catName,
     fireReadTablePreEvent(catName, dbName, tblName);
     List<Partition> ret = null;
     Exception ex = null;
+    RawStore rs = getMS();
     try {
-      checkLimitNumberOfPartitionsByFilter(catName, dbName,
-          tblName, args.getFilter(), args.getMax());
-
       authorizeTableForPartitionMetadata(catName, dbName, tblName);
+      if (isPartitionLimitEnabled()) {
+        // Since partition limit is configured, we need fetch at most (limit + 1) partition names
+        int partitionLimit = MetastoreConf.getIntVar(conf, ConfVars.LIMIT_PARTITION_REQUEST);
+        int requestMax = args.getMax();
+        int max = requestMax < 0 ? partitionLimit + 1 : Math.min(requestMax, partitionLimit + 1);
```

Comment:
if requestMax > 0 && requestMax <= partitionLimit, should we ignore the limit check and fall through to`rs.getPartitionsByFilter` directly?

Reply:
Nice catch, it should be.

Reply:
This logic can also be used in other list partitions api, and now it actually causes a bug, create a follow-up ticket [HIVE-28239](https://issues.apache.org/jira/browse/HIVE-28239).